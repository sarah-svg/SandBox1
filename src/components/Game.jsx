import React from 'react'
import PropTypes from 'prop-types'

function Game({ rock, paper, shears, onClick}) {
    return (
        <form onClick={onClick}>
            <input id="rock"value="rock">{rock}</input>
            <input id="paper"value="paper">{paper}</input>
            <input id="shears" value="shears">{shears}</input>
            <button>Hit</button>
        </form>
    )
}

Game.propTypes = {
    rock: PropTypes.string.isRequired,
    paper: PropTypes.string.isRequired,
    shears: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
}

export default Game

