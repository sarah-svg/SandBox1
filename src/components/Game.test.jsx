import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Game from './Game';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<Game />);
    expect(asFragment()).toMatchSnapshot();
  });
});
