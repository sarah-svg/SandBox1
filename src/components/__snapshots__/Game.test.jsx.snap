// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <form>
    <input
      id="rock"
      value="rock"
    />
    <input
      id="paper"
      value="paper"
    />
    <input
      id="shears"
      value="shears"
    />
    <button>
      Hit
    </button>
  </form>
</DocumentFragment>
`;
